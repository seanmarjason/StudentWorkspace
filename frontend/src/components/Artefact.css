.artefactOverlay {
  z-index: 100;
  position: fixed;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  background-color: white;
  box-shadow: 0 0 25px rgb(0 0 0 / 20%);
  padding: 50px;
  text-align: center;
}

.artefactOverlay .button-container {
  margin-top: 50px;
}

.artefactOverlay .button-container > * {
  border: 1px solid transparent;
  border-radius: 5px;
  background-color: rgb(86, 86, 255);
  padding: 10px 5px;
  margin: 5px;
  font-size: 1em;
  font-weight: bold;
  color: white;
  text-decoration: none;
  text-align: center;
  min-width: 150px;
}

.artefactOverlay .button-container > *:hover {
  cursor: pointer;
  text-decoration: underline;
  box-shadow: 0px 5px 5px 0px rgb(0 0 0 / 25%);
  transform: translate(0, -5%);
}

.artefactOverlay .button-container .download {
  background-color: rgb(123, 243, 90);
}

.artefactOverlay .button-container .open {
  background-color: rgb(86, 224, 255);
}

.artefactOverlay .button-container .delete {
  background-color: rgb(255, 86, 86);
}

