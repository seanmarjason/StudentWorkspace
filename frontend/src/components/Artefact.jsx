import axios from 'axios';
import { useOutsideClick } from '../hooks/useOutsideClick';
import './Artefact.css';

const handleDownloadClick = async (sectionName, filename) => {
  const url = `http://localhost:3000/documents/download/${sectionName}/${filename}`;

  axios({
    url,
    method: 'GET',
    responseType: 'blob',
    // responseType: 'stream' // TODO: Handle response as stream to download bigger files
  })
    .then(response => {
      const link = document.createElement('a');
      link.href = window.URL.createObjectURL(new Blob([response.data]));
      link.setAttribute('download', filename);
      document.body.appendChild(link);
      link.click();
    })
    .catch(error => console.log(`Error: ${error}`))


}

const Artefact = ({ artefact, setArtefact, sectionName }) => {

  const handleClickOutside = () => {
    setArtefact(false);
  };

  const ref = useOutsideClick(handleClickOutside);

  return (
    <div ref={ref} className="artefactOverlay">
      <p>Workspace Artefact</p>
      <p>for artefact: {artefact}</p>
      <span className="closeButton" onClick={() => setArtefact(null)}>&#x2715;</span>
      <div className="button-container">
        <button
          onClick={() =>handleDownloadClick(sectionName, artefact)}
        >
          Download
        </button>
      </div>
    </div>
  )
}

export { Artefact };
