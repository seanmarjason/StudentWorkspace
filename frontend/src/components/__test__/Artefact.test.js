import {render } from '@testing-library/react';
import { Artefact } from '../Artefact';

describe('Artefact' , () => {
  it('should match snapshot for file artefacts', async () => {
    const artefact = {
      "type": "file",
      "name": "name.txt",
      "url": "./uploads/name.txt"
    }

    const setArtefact = jest.fn();

    const sectionName = 'test'

    const view = render(
      <Artefact 
        artefact={artefact} 
        setArtefact={setArtefact} 
        sectionName={sectionName}
      />
    )

    expect(view.asFragment()).toMatchSnapshot();
  });
  
})
  