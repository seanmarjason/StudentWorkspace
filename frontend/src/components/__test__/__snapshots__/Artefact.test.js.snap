// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artefact should match snapshot for file artefacts 1`] = `
<DocumentFragment>
  <div
    class="artefactOverlay artefactOverlay-file"
  >
    <h4>
      Workspace file
    </h4>
    <span
      class="closeButton"
    >
      âœ•
    </span>
    <p>
      name.txt
    </p>
    <div
      class="button-container"
    >
      <button
        class="download"
      >
        Download
      </button>
      <button
        class="delete"
      >
        Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
